### ISSUE TITLE
Customizable default date autoview format

### ISSUE THREAD

**Issue by jmoore34** [nushell/nushell#3735]
**Is your feature request related to a problem? Please describe.**
When performing quick calculations on dates, the default output format is not necessarily the fastest to read. In particular, allowing for a custom default format can allow a user to use a familiar localization as well as break up the different parts with color codes. Compare these two:

![image](https://user-images.githubusercontent.com/1783464/124527221-b3867d80-ddca-11eb-854b-fcf4635602b8.png)

**Describe the solution you'd like**
Perhaps the user could set the default format via a command like:
```
config set set date-format  $"(ansi c)%A (ansi cb)%B %-d(ansi reset)(ansi c), %Y (ansi wb)%-I(ansi g):(ansi wb)%M(ansi reset)(ansi g):(ansi w)%S (ansi w)%p  (ansi blue)[%Z]" 
```
or
```
date set-default-format  $"(ansi c)%A (ansi cb)%B %-d(ansi reset)(ansi c), %Y (ansi wb)%-I(ansi g):(ansi wb)%M(ansi reset)(ansi g):(ansi w)%S (ansi w)%p  (ansi blue)[%Z]" 
```

**Describe alternatives you've considered**
`def` a command like such:
![image](https://user-images.githubusercontent.com/1783464/124527712-dc5b4280-ddcb-11eb-880a-749beefa9fcd.png)
This solution works decently well, though it is a bit more repetitive than changing a config variable.

**Additional context**
One potential issue that might arise if dates are passed along as strings is:
1. The user changes their date format
2. The user passes a date to a script
3. The script changes the date format and then calls other `date` commands on the string (e.g. `date to-table`)

Perhaps this could be avoided if dates were always passed around as strucured data and only rendered as a formatted string at the end of a pipeline (i.e., when passed to an (implicit) autoview).

**Comment by fdncred** [comment]
we would probably accept a PR for something like this but it would be even more cool if there was a way to do it generically where every (or most) commands could participate in this custom coloring output.

**Comment by zoechi** [comment]
I was looking for something like https://www.gnu.org/software/coreutils/manual/html_node/Formatting-file-timestamps.html
where a config variable like `TIME_STYLE` would be used as default and can be overridden per command (if it outputs dates) with a `--time-style` parameter.
In addition it would be nice to be able to configure the time span that determines if the value is recent or not, like for example add seconds if the value is less than 3h old.
I don't care about locale because I exclusively use en-US, but others might.

**Comment by sholderbach** [comment]
Implemented in spirit by #9500
